esp-hal-common/src/lib.rs

```diff
@@ -238,23 +251,31 @@ mod critical_section_impl {
             unsafe fn acquire() -> critical_section::RawRestoreState {
                 let mut mstatus = 0u32;
                 core::arch::asm!("csrrci {0}, mstatus, 8", inout(reg) mstatus);
-                let interrupts_active = (mstatus & 0b1000) != 0;
+                let mut tkn = ((mstatus & 0b1000) != 0) as critical_section::RawRestoreState;
                 #[cfg(multi_core)]
                 {
-                    let guard = multicore::MULTICORE_LOCK.lock();
-                    core::mem::forget(guard); // forget it so drop doesn't run
+                    match super::multicore::MULTICORE_LOCK.lock() {
```

Comment:
Same here, lets document which "reserved bits" were used to smuggle this flag through.

Reply:
`tkn` comes from a bool here, is that still necessary?

Reply:
We enable `critical-section/restore-state-u8` for RISCV, as per the RISCV priv spec (pg 21), bits 6-31 are reserved, so in this case 6-8 are the reserved.